reposilite-backend/src/main/kotlin/com/reposilite/storage/filesystem/FileSystemStorageProvider.kt

```diff
@@ -160,55 +167,48 @@ abstract class FileSystemStorageProvider protected constructor(
-                    contentType = ContentType.getContentTypeByExtension(path.getExtension()) ?: APPLICATION_OCTET_STREAM,
+                    name = path.name,
+                    contentType = ContentType.getContentTypeByExtension(path.extension) ?: APPLICATION_OCTET_STREAM,
                     contentLength = it.size(),
                     lastModifiedTime = it.lastModifiedTime().toInstant(),
                 )
             }.asSuccess()
 
     private fun toDirectoryInfo(directory: Path): Result<DirectoryInfo, ErrorResponse> =
         DirectoryInfo(
-            name = directory.getSimpleName(),
+            name = directory.name,
             files =
-                Files.list(directory).use { directoryStream ->
-                    directoryStream.asSequence()
-                        .map { toSimpleFileDetails(it).orThrow { error -> IOException(error.message) } }
-                        .sortedWith(FilesComparator({ VersionComparator.asVersion(it.name) }, { it.type == DIRECTORY }))
-                        .toList()
-                }
+            directory.useDirectoryEntries { directoryStream ->
+                directoryStream.map { toSimpleFileDetails(it).orThrow { error -> IOException(error.message) } }
+                    .sortedWith(FilesComparator({ VersionComparator.asVersion(it.name) }, { it.type == DIRECTORY }))
+                    .toList()
+            }
         ).asSuccess()
 
     private fun toSimpleFileDetails(file: Path): Result<out FileDetails, ErrorResponse> =
         when (file.type()) {
             FILE -> toDocumentInfo(file)
-            DIRECTORY -> SimpleDirectoryInfo(file.getSimpleName()).asSuccess()
+            DIRECTORY -> SimpleDirectoryInfo(file.name).asSuccess()
         }
 
+    @OptIn(ExperimentalPathApi::class)
     override fun removeFile(location: Location): Result<Unit, ErrorResponse> =
         location.resolveWithRootDirectory().map { rootPath ->
             when {
-                Files.isDirectory(rootDirectory) ->
-                    Files.walk(rootPath).use { directoryStream ->
-                        directoryStream
-                            .sorted(Comparator.reverseOrder())
-                            .map(Path::toFile)
-                            .forEach(File::delete)
-                    }
-                else -> Files.delete(rootPath)
+                rootDirectory.isDirectory() -> rootPath.deleteRecursively()
+                else -> rootPath.deleteExisting()
             }
         }
 
     override fun getFiles(location: Location): Result<List<Location>, ErrorResponse> =
         location.resolveWithRootDirectory()
             .exists()
             .map {
-                Files.walk(it, 1).use { directoryStream ->
-                    directoryStream.asSequence()
-                        .filter { path -> path != it }
+                it.useDirectoryEntries { directoryStream ->
+                    directoryStream.filter { path -> path != it }
```

Comment:
I wonder why it was based on `Files.walk(it, 1)`, not just a `Files.list()` :thinking: